platform/viewer/src/lib/DICOMSR/retrieveDataFromMeasurements.js

```diff
@@ -0,0 +1,52 @@
+import * as dcmjs from 'dcmjs';
+import OHIF from '@ohif/core';
+import cornerstone from 'cornerstone-core';
+
+const retrieveDataFromMeasurements = measurements => {
+  const cornerstoneAdpater = dcmjs.adapters.Cornerstone;
+  const { MeasurementReport } = cornerstoneAdpater;
+  const { getImageIdForImagePath } = OHIF.measurements;
+
+  const toolState = {};
+  const unsupportedTools = [];
+
+  Object.keys(measurements).forEach(measurementType => {
+    const annotations = measurements[measurementType];
+
+    annotations.forEach(annotation => {
+      const { toolType, imagePath } = annotation;
+      if (cornerstoneAdpater[toolType]) {
+        const imageId = getImageIdForImagePath(imagePath);
+        toolState[imageId] = toolState[imageId] || {};
+        toolState[imageId][toolType] = toolState[imageId][toolType] || {
+          data: [],
+        };
+
+        toolState[imageId][toolType].data.push(annotation);
+      } else {
+        unsupportedTools.push(toolType);
+      }
+    });
+  });
+
+  if (unsupportedTools.length > 0) {
+    OHIF.log.warn(
+      `[DICOMSR] Tooltypes not supported: ${unsupportedTools.join(', ')}`
+    );
+  }
+
+  try {
+    const report = MeasurementReport.generateReport(
+      toolState,
+      cornerstone.metaData
+    );
+    return {
+      dataset: report.dataset,
+      unsupportedTools,
+    };
+  } catch (error) {
+    throw error;
```

Comment:
Does that make this a pointless try/catch?

Reply:
I've added this because once we try to save an empty SR (no measurements) `generateReport` throws an error. Did this way, so I can grab the error and display it properly using (as of now) the snackbar.

Reply:
That makes sense, but wouldn't the error already bubble? We're catching and rethrowing?

Reply:
You are right. 
Just fixed here and in another function that I saw the same happening. 

This might be here because I was trying to use the snackbar directly here. Should have removed it after refactor.